src/components/app.js

```diff
@@ -4,39 +4,58 @@ import { bindActionCreators } from 'redux'
 import { createSelector } from 'reselect'
 import ReactTooltip from 'react-tooltip'
 import ActionCreators from '../actions/actionCreators'
+import updateQueryString from '../util/update-query-string.js'
+import removeLocationHash from '../util/remove-location-hash.js'
 
 import Header from './header'
 import Results from './results'
 import SearchBar from './searchbar'
 
 const App = React.createClass({
   componentDidMount() {
-    const query = { ...this.props.query }
+    //
+    // redirect old hash links to query strings
+    //
     const hash = decodeURIComponent(window.location.hash)
     if (hash) {
       const options = hash.slice(1).split(';')
-      const object = {}
       options.forEach(option => {
-        const keyvalue = option.split('=')
-        object[keyvalue[0]] = keyvalue[1]
+        const pair = option.split('=')
+        const key = pair[0]
+        const value = pair[1]
+        updateQueryString(key, value)
```

Comment:
again, we maintain backwards compatability with old, location hash based search links.

you can try an old one against the new code on the dev server like this:

https://1c3aa83f.ngrok.io/search#user=enjalot;d3version=v4

Reply:
@curran notice how the hash is automatically upgraded to a query string when you visit that link ✨

Reply:
Fantastic!